Standards/scs-0112-v1-sonic.md

```diff
@@ -0,0 +1,99 @@
+
+### OVN for SONiC
+
+OVN and OVS are extensively reveraged in Neutron to SDN. In large scale deployments, OVN can become a bottleneck by exhausing resources on network nodes. SONiC can host the OVN control plane as a module (container) and spare the resources in network nodes. There is however a potential other bottleneck on SONiC hardware, as OVN control plane can be resource intensive. This is a potential area for further investigation.
+
+## Design Considerations
+
+There are different ways SONiC support can be implemented in SCS, very similar to existing approaches with Linux.
+
+### Options considered
+
+#### Option 1: SCS distribution of SONiC
+
+With this approach, SCS will create it's own distribution of SONiC, similar to what Debian or Arch are for Linux. This distribution will be based on the SONiC community distribution, but will have SCS specific modules, which will be developed and maintained by SCS. SCS will contribute its code to dedicated SCS repositories and build its own SONiC images. The code can eventually be pushed upstream, but not as top priority. This approach will allow SCS to have a clear path for SONiC support and integration in SCS, but will also require SCS to maintain a distribution of SONiC, which is a significant effort. Upstream/downstream changes will have to be managed and maintained. However the advantage is that SCS will have full control over the distribution and can make changes as needed. Users will have to use the SCS distribution of SONiC, which will be based on the community distribution. If users already deploy community or enterprise SONiC, a migration path to SCS SONiC will be needed.
+
+#### Option 2: SCS will support SONiC but will not change it
+
+SCS supports enterprise ans community versions of SONiC but will not develop its own code for it. This will significantly limit the ability to develop new features for SDN, because all changes will be done in the tooling around SONiC and not in the OS itself. The advantages are that SCS will still improve SONiC support and will have minimal effort for this. The downside is that some features like OVN control plane for SONiC will not be possible.
+
+#### Option 3: SCS develops SCS-specific modules as add-on for any SONiC (Community or Enterprise)
+
+In option 3, SCS will change SONiC by releasing its own modules for it. Those module can be provided as add-ons and installed on top of any version, community or enterprise. While compatability between the modules the SONiC releases will need to be maintained, there will be much broader support for SONiC and users will be able to pick and chose distributions based on their existing relationships and experience and use SCS independent of this. In cases where SCS provides contributions to core SONiC, those can be made in upstream Community repositories, so that the whole community including the propriatory vendors can adopt them eventually.
+
```

Comment:
Could you please add an "Option 4: SCS does not adopt SONiC at all" which outlines the advantages of adopting SONiC in the first place?

Reply:
Thanks @horazont !
Yes, this is definitely a valid option. I will add it to the ADR.